### PR TITLE
merge 'develop_v0.12' into 'release_v0.12'

### PR DESCRIPTION
**Pull request by Lironrad**
Add support for spot percentage under ocean aws launch spec

### CODE REVIEW

File: spotinst/ocean_aws_launch_spec/fields_spotinst_ocean_aws_launch_spec.go [hunk 1]
```diff
@@ -993,6 +994,67 @@ func Setup(fieldsMap map[commons.FieldName]*commons.GenericField) {
...
+			}
+			if result != nil {
+				if err := resourceData.Set(string(Strategy), result); err != nil {
+					return fmt.Errorf(string(commons.FailureFieldReadPattern), string(Strategy), err)
+				}
+			}
+			return nil
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			LaunchSpecWrapper := resourceObject.(*commons.LaunchSpecWrapper)
```
Comment (liranp) [inline spotinst/ocean_aws_launch_spec/fields_spotinst_ocean_aws_launch_spec.go:1030]:
```suggestion
			launchSpecWrapper := resourceObject.(*commons.LaunchSpecWrapper)
```

File: spotinst/ocean_aws_launch_spec/fields_spotinst_ocean_aws_launch_spec.go [hunk 2]
```diff
@@ -993,6 +994,67 @@ func Setup(fieldsMap map[commons.FieldName]*commons.GenericField) {
...
+			launchSpec := LaunchSpecWrapper.GetLaunchSpec()
```
Comment (liranp) [inline spotinst/ocean_aws_launch_spec/fields_spotinst_ocean_aws_launch_spec.go:1031]:
```suggestion
			launchSpec := launchSpecWrapper.GetLaunchSpec()
```

File: spotinst/ocean_aws_launch_spec/fields_spotinst_ocean_aws_launch_spec.go [hunk 3]
```diff
@@ -993,6 +994,67 @@ func Setup(fieldsMap map[commons.FieldName]*commons.GenericField) {
...
+				if strategy, err := expandStrategy(value); err != nil {
+					return err
+				} else {
+					launchSpec.SetStrategy(strategy)
+				}
+			}
+			return nil
+		},
+		func(resourceObject interface{}, resourceData *schema.ResourceData, meta interface{}) error {
+			LaunchSpecWrapper := resourceObject.(*commons.LaunchSpecWrapper)
```
Comment (liranp) [inline spotinst/ocean_aws_launch_spec/fields_spotinst_ocean_aws_launch_spec.go:1042]:
```suggestion
			launchSpecWrapper := resourceObject.(*commons.LaunchSpecWrapper)
```

File: spotinst/ocean_aws_launch_spec/fields_spotinst_ocean_aws_launch_spec.go [hunk 4]
```diff
@@ -993,6 +994,67 @@ func Setup(fieldsMap map[commons.FieldName]*commons.GenericField) {
...
+			launchSpec := LaunchSpecWrapper.GetLaunchSpec()
```
Comment (liranp) [inline spotinst/ocean_aws_launch_spec/fields_spotinst_ocean_aws_launch_spec.go:1043]:
```suggestion
			launchSpec := launchSpecWrapper.GetLaunchSpec()
```

File: spotinst/ocean_aws_launch_spec/fields_spotinst_ocean_aws_launch_spec.go (hunk 1 above)
Comment (Lironrad) [reply spotinst/ocean_aws_launch_spec/fields_spotinst_ocean_aws_launch_spec.go]: @liranp I changed the LaunchSpecWrapper name to launchSpecWrapper across all fields_spotinst_ocean_aws_launch_spec.go file FYI

**Review by liranp — APPROVED** [review]

**Pull request merged**